### DISCUSSION TITLE
Got fpGUI working on Android via Termux/X11/XFCE4 but some language features and fonts do not work..

### DISCUSSION THREAD

**Discussion by PTz0uAH** [graemeg/fpGUI#150] [General]
Hoi @graemeg , thank you for having this repository..

I managed to get fpGUI working with Termux/X11/XFCE4/GTK2 but cannot view the UTF8 mode fonts..

Tried to install Liberation font and other fonts but it is not shown correctly..

The (minimal) needed changes and font-try-outs I made are committed to the fork at: https://github.com/PTz0uAH/fpgui

![Screenshot_20241102-113015](https://github.com/user-attachments/assets/6a67d049-d1fd-43a1-bda2-d517730e75c8)

I know Termux/Android (linux alike approach) is probably not a fully supported platform (`uname -a` shows Linux localhost..etc. but also aarch64 Android) but some advice is welcome..

Anyway.. I thought it would be very cool to get fpGUI fully working, ported to this relatively new and very handy platform.. running on any modern Android device via the Termux-app..

To achieve all this I needed to create FPC/LAZBUILD for Termux manually to create with lazbuild a native build/ported FPCUPDELUXE and a patched fpc/laz source first.. to build a fully working FPC LAZARUS environment.. of which I am using that fresh FPC 3.3.1 build to get fpGUI working..

with kind regards,

Peter

p.s. on WSL2 the example shows how it should look I suppose.. but the window handling and icon are better on the termux version..

![FpGUIonWSL](https://github.com/user-attachments/assets/225933d0-8fda-4b54-a13c-b5fff94aac24)

**Comment by graemeg** [comment]
Hi Peter. So sorry for the delay in response. That is amazing work. Unfortunately I have no capacity or hardware to implement and test such a platform. But do feel free to update us on your progress. Thank you for sharing your work with us. :pray:
